### DISCUSSION TITLE
Crazyflie 2.1 with solid blue M3 LED and no boot

### DISCUSSION THREAD

**Discussion by andrew-t97** [bitcraze/discussions#1587] [Q&A]
## The problem

Since following the [soldering guide ](https://www.bitcraze.io/documentation/tutorials/solder-deck-pins/) and soldering some wires to my crazyflie, I've been experiencing an issue where the crazyflie can't boot and gets stuck with the M3 LED solid blue. Initially, this problem occurred intermittently when I moved a wire around. However, after I removed the problematic wire, the issue seems to occur all the time.

Thanks for any help you can provide ❤️

## What I can currently do
- Turn on the crazyflie to the stage before the bootloader (solid M3 LED)
- Enter bootloader mode on the crazyflie (Only the M3 LED flashes, no other LEDs are lit)
- Manually flash the DFU
- Almost update the firmware via the cfclient (appears to skip the firmware upgrade)
- Flash the firmware via the repo (over the air)
- Charge the battery

## Logs I receive in the cfclient during bootloading
```
INFO:cfclient.ui.dialogs.bootloader:Created temporary firmware-release file at/var/folders/8y/z5vl00kd28jcvyczn00f72jr0000gn/T/tmpm3_x0ugt/firmware-cf2-2024.2.zip
Found manifest version: 2
nRF51 has: sd-s130 and requires sd-s130 and upgrade provides sd-s130. Current bootloader version is [2024.2.0] but upgrade provides [2024.2]
Should flash nRF soft device
Closing bootloader link and reconnecting to new bootloader (radio://0/0/2M/B13747D94F?safelink=0)
Reconnected to new bootloader
Skipping updating deck on coldboot
```

## What I've attempted

I've been following [this](https://github.com/orgs/bitcraze/discussions/172) discussion and those within it to see if I was experiencing a similar problem, however, it hasn't seemed to have helped.

- I've charged the battery
- I've reflashed the firmware via the repo and via the CFClient. My repo is at commit `7f1f78251fccdb355d672d579997238258e367d2`
- I've reflashed via the cfclient
- I've flashed the DFU and then tried reflashing via the CFClient

### Top

![image](https://github.com/user-attachments/assets/10868221-82ec-4871-9252-1c79c936a8c5)

### Bottom

![image](https://github.com/user-attachments/assets/8abc3c1b-dd9d-4ec7-a400-5bfd0c926ff2)

**Comment by tobbeanton** [comment]
* I assume the power button works? The M2 led should lit up so either something is shorted on the nRF-VCC rail or the nRF51 is not flashed as it should
* Is something getting hot while the battery is connected?

My guess it that you have a "bit of a short" somewhere. Do you have isopropanol or other electrical cleaning fluid to clean it with?

**Reply by andrew-t97** [reply]
>I assume the power button works? The M2 led should lit up so either something is shorted on the nRF-VCC rail or the nRF51 is not flashed as it should

The power button is working fine. I've cleaned the board up and checked for shorts, but I can't seem to spot any myself. Below are new pictures of the crazyflie after cleaning. Is there anything suspicious looking on the board?

![PXL_20241009_171047266 RAW-01 COVER](https://github.com/user-attachments/assets/796356d6-6627-4e9a-b4ff-323851351cfe)
![PXL_20241009_171302424 RAW-01 COVER](https://github.com/user-attachments/assets/89a164e9-33be-414a-85b3-9bf07bbe0b67)

>Is something getting hot while the battery is connected?

The only thing that seems to warm up is the STM32.

I'll try flashing the nRF51 now that I've cleaned the board just in case there was a short and it was causing an issue with nRF51 flashing.

**Reply by andrew-t97** [reply]
I've done the following in terms of re-flashing but haven't had any luck:

1. Reflashing the crazyflie via the cfclient
2. Manually reflashing the DFU
3. Reflashing the crazyflie via the firmware repo

**Reply by tobbeanton** [reply]
Can you measure the VCCNRF voltage.
![image](https://github.com/user-attachments/assets/9f4b0af9-9330-4427-864a-08027a63ce57)
The bottom pad is pin1
![image](https://github.com/user-attachments/assets/3d61d14a-b1ed-4340-b753-b465e6c193eb)
So measure between pin the two middle pads

**Reply by andrew-t97** [reply]
Yep! I get 3v across the two middle pads 👍

**Reply by tobbeanton** [reply]
Could there be a short between the top pad and PA5? Try to inspect all your previous solder pads and look for any short.

**Reply by andrew-t97** [reply]
I've gone over all of my previous solder points and can't seem to find any shorts. I've tried to get a better picture of the pads. I don't think there's a short there. What do you think?

![PXL_20241011_115918631 RAW-01 COVER](https://github.com/user-attachments/assets/99c6722e-6378-4892-a99b-e0e0b24db32a)
![PXL_20241011_120259780 RAW-01 COVER](https://github.com/user-attachments/assets/c52088ad-abc1-40fd-82cd-a29a608357b7)

One thing I've noticed is the bits of track revealed over the PA7 label. Could this be an issue? Is there anything else I can check?

I haven't flashed since resoldering the joints. Do you think that's worth doing?

**Reply by tobbeanton** [reply]
As long as there is no short the bits of revealed tracks doesn't matter. Check with a multimeter in resistance mode that there is no short between any of these pins
![image](https://github.com/user-attachments/assets/8e2bf53b-4735-4f98-b835-0de5e223856f)

> I haven't flashed since resoldering the joints. Do you think that's worth doing?

Yes, this might give you more clues.

**Reply by andrew-t97** [reply]
Which pins are you referring to? Between PA6 and PA7 or between the NRF pads?

I tried a reflash, but this didn't seem to change anything. 😕

**Reply by tobbeanton** [reply]
I'm sorry but I don't think we will be able to solve this. It is just to complicated to troubleshoot.

**Reply by andrew-t97** [reply]
No worries @tobbeanton, thanks for all your help! Do you know if there's anyone else that might be able to help me?

**Reply by varadVaidya** [reply]
Have you been able to solve this issue? I am also facing the a similar issue, but rather a very low dim light. The light sometimes doesn't even turn on. Although i seem to be able to enter the firmware mode, and able to fly. However it tends to crash right after takeoff. 

I have also attempted all the troubleshooting steps you mentioned to no avail. I am hoping to get a detailed log and request solution in a new thread. But till then would love to hear if you were able to solve this particular problem.

**Reply by andrew-t97** [reply]
Hi @varadVaidya, hmmm, this seems like quite a different issue from the one I raised originally. I couldn't resolve this issue and purchased a new CF2.1 board to get around it. It sounds like your issue might have more hope, as it's a good sign that you can fly.

**Reply by varadVaidya** [reply]
@andrew-t97, thanks for your response. I should have updated the post. 
The crazyflie is now seemingly working perfectly fine. I checked it after a day or so, and no signs of any problems.
